### PR TITLE
Create our own Enumerable

### PR DESCRIPTION
**Pull request by RolandM99**
In this PR we performed the following :

- Use Ruby syntax for basic programming operations.
- Apply Ruby best practices and language style guides in code.

### CODE REVIEW

File: my_enumerable.rb  [hunk 1]
```diff
@@ -0,0 +1,35 @@
+module MyEnumerable
+  def list
+    @list ||= []
+  end
+
+  def all?
+    return unless block_given?
```
Comment (oloomoses) [inline my_enumerable.rb ]: - [ ] `my_enumerable.rb` has a trailing white space and therefore generates an error. Kindly fix it.

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require_relative 'my_enumerable'
+
+class MyList
+  include MyEnumerable
+  def initialize(*list)
+    @list = list
+  end
+end
```
Comment (oloomoses) [inline main.rb]: - [x] Kindly add `#each` method in `MyList` class. In MyList implement a method #each that yields successive members of @list and uses the MyEnumerable module.
the code should resemble this: 
```
def each
    return to_enum(:each) unless block_given?

    counter = 0
    while counter < @list.length
      yield(@list[counter])
      counter += 1
    end
    @list
end
```
- [x] The initialize class should look like below:
```
def initialize(*args)
        @list = []
        args.each { |arg| @list << arg }
    end
```

**Review by oloomoses — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 2 above)
Comment (aakbarkhan) [reply main.rb]: Implement the changes, thank you.

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**